### ISSUE TITLE
asap-prep does not support --oe-active-site-residue:

### ISSUE THREAD

**Issue by SiyangPeng** [choderalab/asapdiscovery#927]
I have been trying to run command line but : 
asap-prep --target ZIKV-NS2B-NS3pro --pdb-file zikv_ns3.pdb --output-dir orig_prepped --align zikv_ns3.pdb --oe-active-site-residue TYR:161::B --ref-chain B --active-site-chain B

The error log indicates: 
Warning: Failed, no pocket found with site residue predicate
Warning: Unable to parse and utilize specified site residue TYR:161::B to define site in _chainB_ASP-MET-TYR-ILE-GLU-ARG-ALA-GLY-ASP-ILE-THR-TRP-GLU-LYS-ASP-ALA-GLU-VAL-THR-GLY- - skipping
ValueError: Failed to make design unit.

I identified the active site residue by running this code block using the get_interactions_plip function from asapdiscovery plip (https://github.com/choderalab/asapdiscovery/blob/main/asapdiscovery-data/asapdiscovery/data/backend/plip.py#L214):
protein_orig = load_openeye_pdb('/home/pengs/fold_zika/test_prep/zikv_ns3.pdb')
pose_orig = load_openeye_sdf('/home/pengs/fold_zika/lig_sdfs/lig_0.sdf')
color_method = 'fitness'
target = "ZIKV-NS2B-NS3pro"
get_interactions_plip(protein_orig, pose_orig, color_method, target)

I think the reason why this is failing is because I am not able to identify the residue identified by plip with the oeGraphMol object constructed during prepping and so it could not identify a binding pocket to dock the ligand even though there is a clear pocket there.

**Comment by hmacdope** [comment]
@SiyangPeng can I suggest you try an make a minimal example to replicate this that uses **just** the raw underlying openeye functionality ie `OEMakeDesignUnits`  oespruce.OEMakeDesignUnits(initial_prot, metadata, opts, site_residue)

Forget about PLIP for now, is there a tyrosine at position 161 in the input protein? 

Also check carefully that you are using the right OpenEye syntax for the residue. This issue happens most often when you get that very slightly wrong (this has happened to me a lot).

**Comment by SiyangPeng** [comment]
I have tried to do that. It did not make a designunit. The error message was 
`Warning: No BioAssembly transforms found, using input molecule as biounit: 
Warning: Iridium - Structure:  has no REMARK data
Processing BU # 1 with title: , chains AB
Warning: Failed, input predicate does not match any atom in molecule _(AB)
Warning: Failed, no pocket found with site residue predicate
Warning: Unable to parse and utilize specified site residue TYR:1::B to define site in  - skipping
Warning: No BioAssembly transforms found, using input molecule as biounit: `

There is a tyrosine at position 161 in the input PDB file. I loaded the input pdb file into pymol and there was a tyrosine at resid 161 and it seemed like a reasonable place for a pocket. 

I think I am using the right syntax. I have been using this as reference: https://github.com/choderalab/asapdiscovery/blob/1ec8acf2de9753ccbc311dcdaeb8376e35b8a4b0/asapdiscovery-modeling/asapdiscovery/modeling/modeling.py#L186

**Comment by SiyangPeng** [comment]
> Warning: Unable to parse and utilize specified site residue TYR:1::B to define site in - skipping Warning: No BioAssembly transforms found, using input molecule as biounit:
> 
> This would suggest the syntax is not being parsed, try an example from the openeye manual directly. :)

I have been trying to find a manual to reference, but the closest I have found is from openforcefield topology (https://docs.openforcefield.org/projects/toolkit/en/0.5.1/_modules/openforcefield/topology/topology.html)

I tried to find the residues in OEGraphMol prot by running this:
`hv = oechem.OEHierView(
            prot, oechem.OEAssumption_BondedResidue +
            oechem.OEAssumption_ResPerceived + oechem.OEAssumption_PDBOrder)
topology = app.Topology()

for chain in hv.GetChains():
    for frag in chain.GetFragments():
        for hres in frag.GetResidues():
            # Get OE residue
            oe_res = hres.GetOEResidue()
            print(str(oe_res))`

The residue seem to be stored in this format 'TYR 161   B 2'

I put this into `du = oespruce.OEMakeDesignUnits(prot, metadata, opts, 'TYR 161   B')` but it is still giving me the same error.

**Comment by hmacdope** [comment]
Try ":161" or something similar. @apayne97 will have used these options as well. I am 90% sure we are just getting syntax wrong.

**Comment by apayne97** [comment]
Yes, I still need to follow up on trying this. I'll be putting a jupyter notebook with my attempt here: https://github.com/choderalab/fragment-fold/tree/main/science/2024.03.29_test_apo_prep

**Comment by apayne97** [comment]
Ok so....unfortunately I think the problem here is that the code for doing this was dismantled and moved right before version 0.1 and I went to Schrödinger and we focused on other things and we've kind dropped this.

https://github.com/choderalab/asapdiscovery/pull/238/files

I saved some minimal examples here: https://github.com/choderalab/asapdiscovery-mers-docking/tree/main

And I think the reality is that I never actually got the active site residue to thing to really work consistently (as Siyang has found with the fragment screen stuff). 

Instead I think in order to generate our "foldalysis" database, what we want to do is a more literal copy-and-paste of ligand coordinates in order to generate DUs a la:
https://github.com/choderalab/asapdiscovery-mers-docking/blob/main/mers_fauxalysis_generation/scripts/copy_ligands_to_new_structures.py

Followed by a secondary self-docking of each fragment into its own DU to allow for the ligand to find a pose that is as close to the original as possible while avoiding clashes with the protein.
https://github.com/choderalab/asapdiscovery-mers-docking/blob/main/mers_fauxalysis_generation/scripts/run_self_docking_oe.py

I think I'll add this functionality as a new ProteinPrepper? What do you think @hmacdope

**Comment by hmacdope** [comment]
@apayne97 if you have a solution that provides a way forward, go for it.

**Comment by apayne97** [comment]
For posterity, this syntax "worked": 
`"TYR:161: :B:0: "`
But this did not:
`"TYR:161: :A"`

In any case, it only worked in the sense that it gave me this error:
```
"Warning: Failed to find a pocket associated with the input predicate\n",
"Warning: Failed, no pocket found with site residue predicate\n",
"Warning: Unable to parse and utilize specified site residue TYR:161: :B:0:  to define site in  - skipping\n"
```

Instead of this error:
```
"Warning: Failed, input predicate does not match any atom in molecule _(AB)\n",
"Warning: Failed, no pocket found with site residue predicate\n",
"Warning: Unable to parse and utilize specified site residue TYR:161: :A to define site in  - skipping\n",
```

And I think this *might* be because we're using a fragment screen so it has a hard time finding normal ligand-like binding pockets. 

So I'm doing #972 instead.

**Comment by apayne97** [comment]
We still should update our docs to reflect the correct active site syntax (or maybe delete it).
